### PR TITLE
feat: intial support for binary_sensor devices

### PR DESCRIPTION
**Pull request by Jc2k**
Just adds a couple of sensors so can start bootstrapping things in HA.

### CODE REVIEW

**Comment by Jc2k** [comment]
@Ernst79 do you know if obj1014 and obj1015 send on/off messages or do they depend on the timer?

File: src/xiaomi_ble/parser.py [hunk 1]
```diff
@@ -525,7 +534,9 @@ def obj1007(
         if device_type in ["MJYD02YL", "MCCGQ02HL"]:
             # 100 means light, else dark (0 or 1)
             # MCCGQ02HL might use obj1018 for light sensor, just added here to be sure.
-            return {"light": 1 if illum == 100 else 0}
+            device.update_predefined_binary_sensor(
+                BinarySensorDeviceClass.LIGHT, True if illum == 100 else False
```
Comment (bdraco) [inline src/xiaomi_ble/parser.py:538]: This could be wrapped in bool instead

Comment (Jc2k) [reply src/xiaomi_ble/parser.py:538]: Or indeed `illum == 100` would return a bool anyway.

**Review by bdraco — APPROVED** [review]

**Comment by Ernst79** [comment]
> @Ernst79 do you know if obj1014 and obj1015 send on/off messages or do they depend on the timer?

Ill try to check tonight.

**Comment by Jc2k** [comment]
Tyvm!

**Comment by Ernst79** [comment]
obj1015 is a binary sensor (0 = no smoke, 1 = smoje detected).

Obj1014 is a binary sensor for SJWS01LM flooding sensor. It reports 0 = dry, 1 = wet. 

Both do NOT need a timer.

**Comment by Jc2k** [comment]
Awesome!

**Pull request merged**